Update for new asset URLS in recent argo releases

danajp opened the pull request:
In recent releases of argo, they've started gzipping their release assets before uploading them. See, for example, this release https://github.com/argoproj/argo/releases/tag/v2.10.1.

This change tries to download from the gz urls and falls back on the non-gz urls if the assets aren't found. This assumes they'll continue to gzip assets and that people are mostly going to want newer releases of argo.

The second commit just makes this code compliant with [shellcheck](https://github.com/koalaman/shellcheck).

Let me know if you want me to make any changes. Thanks for maintaining this plugin!

danajp committed:
Assets are gzipped in recent releases

In releases 2.10+, assets are gzipped. Try the new .gz urls before
falling back on the old uncompressed urls.

danajp committed:
Make this code shellcheck compliant

sudermanjr commented:
@danajp  Thanks, this is awesome. Looks like the build is failing due to Argo wanting a kube config when checking version? (I'm guessing at first glance) That seems like an Argo problem

```
  time="2020-09-04T16:54:25Z" level=fatal msg="invalid configuration: no configuration has been provided"
  FAILED: argo version failed with exit code 0
```

sudermanjr commented:
I filed an issue about this - https://github.com/argoproj/argo/issues/3943

danajp commented:
The purpose of the test is just to make sure we've got an argo binary that runs, right? Maybe we can change the test command to `argo -h`. That doesn't seem to care if we've got a kube config or not.

danajp committed:
Change test command to `argo -h`

Recent versions of argo fail if no kube config exists, even for `argo version`.

sudermanjr commented:
> The purpose of the test is just to make sure we've got an argo binary that runs, right? Maybe we can change the test command to `argo -h`. That doesn't seem to care if we've got a kube config or not.

Good call. For some reason I thought there was version checking, but there is apparently not. Thanks!
